src/Sentry/SentryOptions.cs

```diff
@@ -1182,6 +1181,7 @@ public bool JsonPreserveReferences
     public SentryOptions()
     {
         SettingLocator = new SettingLocator(this);
+        _lazyInstallationId = new(() => new InstallationIdHelper(this).TryGetInstallationId());
```

Comment:
And because it's `readonly` now it's ensured to be set only once in the future as well.

Reply:
The main problem before was actually this:
```csharp
    private Lazy<string?> LazyInstallationId => new(() => new InstallationIdHelper(this).TryGetInstallationId());
```

It's subtle but that was actually declaring a method that creates a new Lazy (not a get only property), so every time you call it a new InstallationIdHelper gets created and nothing gets cached. 

The new code declares a backing field and assigns it to a (readonly) lazy instance. Since it's using non-static fields/properties in the initialization, it has to go in the constructor. But yeah, the readonly is nice reassurance.